File: 2022B/a.cpp
#include <bits/stdc++.h>
using namespace std;

double area(int radius)
{
    return (radius * radius * 3.1415927);
}

void solve()
{
    int R, A, B;
    double res = 0;
    cin >> R >> A >> B;
    for (int i = 0; i < 5763; i++, R *= A, R /= B)
        res += area(R) + area(R * A);
    cout << fixed << setprecision(6) << res << "\n";
}

int main(void)
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int T;
    cin >> T;
    for (int i = 1; i <= T; i++)
    {
        cout << "Case #" << i << ": ";
        solve();
    }
    return 0;
}

File: 2022B/a.py
import math


def area(radius): return radius**2 * 3.1415927


def solve():
    R, A, B = [int(x) for x in input().split()]
    n = int(math.log(R, B/A))
    res = 0
    for i in range(n+1):
        res += area((R * ((A / B)**i))) + \
            area((R * ((A ** (i + 1)) / (B ** i))))
    print('{0:.6f}'.format(res))


if __name__ == "__main__":
    T = int(input())
    for i in range(1, T+1):
        print(f'Case #{i}: ', end="")
        solve()
